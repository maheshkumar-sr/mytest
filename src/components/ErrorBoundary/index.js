/* @flow */
import React from 'react';
import PropTypes from 'prop-types';

interface Props {
  children: Object;
}

interface State {
  error: any;
  errorInfo: any;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { error: null, errorInfo: null };
  }

  componentDidCatch(error: any, errorInfo: any) {
    // Catch errors in any components below and re-render with error message
    this.setState({
      error,
      errorInfo
    });
    // You can also log error messages to an error reporting service here
  }

  render() {
    const { state } = this;
    const { children } = this.props;

    if (state.errorInfo) {
      // Error path
      return (
        <>
          <h2>Something went wrong.</h2>
          <details style={{ whiteSpace: 'pre-wrap' }}>
            {state.error && state.error.toString()}
            <br />
            {state.errorInfo.componentStack}
          </details>
        </>
      );
    }
    // Normally, just render children
    return children;
  }
}
ErrorBoundary.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
};

export default ErrorBoundary;
