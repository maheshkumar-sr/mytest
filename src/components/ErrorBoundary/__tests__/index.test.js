import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorBoundary from '..';

describe('<ErrorBoundary /> ', () => {
  let PROPS = {};
  beforeEach(() => {
    PROPS = {
      angled: false,
      leftArrow: false,
      rightArrow: false,
      discreet: true,
      fullwidth: 'true',
      children: '',
      variant: '',
      emphasis: '',
      customClass: {},
      theme: 'black',
      fullWidth: true
    };
  });

  test('render the component correctly', () => {
    const wrapper = mount(<ErrorBoundary {...PROPS} />);
    expect(wrapper.exists()).toBe(true);
  });

  test('render the component correctly', () => {
    const wrapper = mount(<ErrorBoundary {...PROPS} />);
    wrapper.setState({
      error: null,
      errorInfo: null
    });
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('componentDidCatch with null params', () => {
    const wrapper = shallow(<ErrorBoundary {...PROPS} />);

    const error = null;
    const errorInfo = null;

    wrapper.instance().componentDidCatch(error, errorInfo);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('componentDidCatch without null params', () => {
    const wrapper = shallow(<ErrorBoundary {...PROPS} />);

    const error = 'error';
    const errorInfo = 'componentDidCatch';

    wrapper.instance().componentDidCatch(error, errorInfo);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
