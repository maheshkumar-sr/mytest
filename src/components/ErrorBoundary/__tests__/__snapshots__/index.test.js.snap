// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary />  componentDidCatch with null params 1`] = `""`;

exports[`<ErrorBoundary />  componentDidCatch without null params 1`] = `
<Fragment>
  <h2>
    Something went wrong.
  </h2>
  <details
    style={
      Object {
        "whiteSpace": "pre-wrap",
      }
    }
  >
    error
    <br />
  </details>
</Fragment>
`;

exports[`<ErrorBoundary />  render the component correctly 1`] = `
<ErrorBoundary
  angled={false}
  customClass={Object {}}
  discreet={true}
  emphasis=""
  fullWidth={true}
  fullwidth="true"
  leftArrow={false}
  rightArrow={false}
  theme="black"
  variant=""
/>
`;
